src/main/groovy/val/Idator.groovy

```diff
@@ -77,6 +69,58 @@ class Idator implements Check {
         using(definition)
     }
 
+    // Runtime metaprogramming to dispatch to Checkers
+
+    // Used to call to checkers while possibly augmenting the mold with
+    // information from the present context
+    // Will presently most likely have issues if Maps are non-mold parameters
+    // This could be avoided through some form of counting or an annotation
+    // The solution is deferred until the problem arises
+    def moldyCall = { name ->
+        { Object[] args ->
+            // if no args were passed just call with a mold based on the context
+            if (!args) {
+                checkers."${name}"(key:delegate.resultKey)
+            }
+            // if the first arg is a map, assume it is the mold
+            else if (args[0] instanceof Map)  {
+                args[0] = [key:delegate.resultKey] + args[0]
+                checkers."${name}"(*args)
+            }
+            // else the optional mold was not provided so add one as first arg
+            else {
+                checkers."${name}"([key:delegate.resultKey], *args)
+            }
+        }
+    }
+    def methodMissing(String name, args) {
+        def types = args.collect{it.class}
+        def check
+        // If the invocation omitted an optional mold
+        if (checkers.metaClass.respondsTo(checkers, name, Map, *types)) {
+            check = moldyCall(name)
+        }
+        // If the mold was not omitted or not expected
+        else if (checkers.metaClass.respondsTo(checkers, name, *types)) {
+            // If mold (or similar) is present, augment it
+            if (args && args[0] instanceof Map) check = moldyCall(name)
```

Comment:
I just submitted tweaked the code a bit to try to make it a bit more self-documenting